### PR TITLE
Safer Solana staking

### PR DESCRIPTION
**Pull request by sponomarev**
### 📝 Description

As described in the [support article](https://support.ledger.com/hc/en-us/articles/12568029655453-Fixing-a-Sorry-internet-seems-to-be-down-error-with-Solana?support=true), Solana users faces the issue with not enough funds to unstake their SOL pretty often.  To make the process safer this PR adds a friendly warning to reserve some SOL to cover future unstaking fees.

After a small research, we found out that:
- Everstake recommends to keep at least 0.05 SOL
- Phantom also recommends  0.05 SOL
- Solflare recommends 0.01 but keeps 0.05 when “Use Max” is involved

Therefore, we think that 0.01 is a reasonable amount. 

In addition, we found and fixed the discrepancy between maximum spendable amount in the upper text and in the amount field when “Use Max” is selected.

image

Besides that, as requested by the LL team, those requirements were implemented:

- If a user use the "delegate" max feature, the logic should automatically compute a maximum value that takes into consideration the Rent value and the undelegation fees.

- If a users tries to manually input a value that is above the "delegate max" value, then he should be prevented to do so (continue button is greyed out) and  red error message should be displayer "Sorry, insufficient funds"

- If a user has one or more active delegations then, in the send flow, when using the send max feature, the logic should compute a value that takes into consideration the Rent and the fees necessary to undelegate.

As a result, all unstaking reserves are not available for spending on any operations except withdrawing/undelegating. This should make the process safer and easier for end users

### ❓ Context

- **Impacted projects**: LLD Solana users
- **Linked resource(s)**:
  - https://ledgerhq.atlassian.net/browse/LIVE-8292
  - https://ledgerhq.atlassian.net/browse/LIVE-7150

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

image

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Review by haammar-ledger — COMMENTED** [review]
Looks all good, can you please just add a changeset, using `pnpm changeset`? Thanks

**Review by hedi-edelbloute — COMMENTED** [review]
Hey @sponomarev, thanks for your PR

Looks good !

Could we also prevent the user from signing the transaction with a disabled button if he hasn't enough SOL remaining to execute staking operations ? It would be perfect, or a warn message atleast :thinking: 

Your PR is preventing users for ending up in the 0 SOL state in the first place :ok_hand:

**Comment by sponomarev** [comment]
> Looks all good, can you please just add a changeset, using `pnpm changeset`? Thanks

@haammar-ledger I've done that already and it generated a new file. Is anything else needed?

https://github.com/LedgerHQ/ledger-live/pull/5228/commits/20f41cac5f33a33f75a0c9c61b3fd59b3292c3e7#diff-1e2954ab7246c06cc7b0349c22a55060533f619c0c5c2c5bf351056f02db4932R1-R6

**Comment by sponomarev** [comment]
> Hey @sponomarev, thanks for your PR
> 
> Looks good !
> 
> Could we also prevent the user from signing the transaction with a disabled button if he hasn't enough SOL remaining to execute staking operations ? It would be perfect, or a warn message atleast 🤔
> 
> Your PR is preventing users for ending up in the 0 SOL state in the first place 👌

@hedi-edelbloute Would it be to defensive to restrict users to use all of their SOL? I do believe that a huge amount of the audience knows exactly whet they're doing.

**Comment by hedi-edelbloute** [comment]
> > Hey @sponomarev, thanks for your PR
> > Looks good !
> > Could we also prevent the user from signing the transaction with a disabled button if he hasn't enough SOL remaining to execute staking operations ? It would be perfect, or a warn message atleast 🤔
> > Your PR is preventing users for ending up in the 0 SOL state in the first place 👌
> 
> @hedi-edelbloute Would it be to defensive to restrict users to use all of their SOL? I do believe that a huge amount of the audience knows exactly whet they're doing.

Hey @sponomarev , what I mean is, on ledger live, we prevent the user from attempting transactions that will fail, for example when you try to send more than what you have

Here when the user hasn't enough SOL to pay for fees (for withdrawing/unstaking), we could disable the button the user cannot even broadcast the tx

**Comment by hedi-edelbloute** [comment]
Hey @sponomarev , thanks for your update, i'm going to test it manually

**Comment by hedi-edelbloute** [comment]
hey @sponomarev , I did a "use max" on a delegation and ended up stuck on this frame when using deactivate

image

**Comment by sponomarev** [comment]
> hey @sponomarev , I did a "use max" on a delegation and ended up stuck on this frame when using deactivate

@hedi-edelbloute This is expected. I believe we should not restrict users from disposing of their property. On the staking step we just warn them regarding the reserve to cover future fees. Use Max simply calculates max available SOL to make the transaction successful. After Use Max only main account rent fee is left. Doesn't it look correct to you?

**Comment by hedi-edelbloute** [comment]
> > hey @sponomarev , I did a "use max" on a delegation and ended up stuck on this frame when using deactivate
> 
> @hedi-edelbloute This is expected. I believe we should not restrict users from disposing of their property. On the staking step we just warn them regarding the reserve to cover future fees. Use Max simply calculates max available SOL to make the transaction successful. After Use Max only main account rent fee is left. Doesn't it look correct to you?

@sponomarev wasn't it supposed to keep 0.01 SOL when user uses "use max" toggle ?

**Comment by sponomarev** [comment]
@hedi-edelbloute  Nope, it does not.

> > > hey @sponomarev , I did a "use max" on a delegation and ended up stuck on this frame when using deactivate
> > 
> > 
> > @hedi-edelbloute This is expected. I believe we should not restrict users from disposing of their property. On the staking step we just warn them regarding the reserve to cover future fees. Use Max simply calculates max available SOL to make the transaction successful. After Use Max only main account rent fee is left. Doesn't it look correct to you?
> 
> @sponomarev wasn't it supposed to keep 0.01 SOL when user uses "use max" toggle ?

@hedi-edelbloute Nope, it does not. It's just a friendly warning.

**Comment by hedi-edelbloute** [comment]
@sponomarev FYI my coworker shared product specs here : https://discord.com/channels/885256081289379850/921270433070841856/1172113705740283984

Thanks :pray:

**Comment by sponomarev** [comment]
> @sponomarev On mobile the user doesn't know why he cannot deactivate if he has insufficient funds, could you also add the information there ?
> 


@hedi-edelbloute It took a while to set local mobile development environment up and running with hardware ledger. Nevertheless, I did it =)

Here are some screenshots for the changes you requested

Simulator Screenshot - iPhone SE (3rd generation) - 2023-11-14 at 14 23 33

Simulator Screenshot - iPhone SE (3rd generation) - 2023-11-14 at 14 22 50

**Comment by sponomarev** [comment]
> @sponomarev FYI my coworker shared product specs here : https://discord.com/channels/885256081289379850/921270433070841856/1172113705740283984
> 
> Thanks 🙏

Unfortunately, I'm not able to join this channel. I assume it's private. Will you be able to add me there? 

image

**Comment by hedi-edelbloute** [comment]
> > @sponomarev FYI my coworker shared product specs here : https://discord.com/channels/885256081289379850/921270433070841856/1172113705740283984
> > Thanks 🙏
> 
> Unfortunately, I'm not able to join this channel. I assume it's private. Will you be able to add me there?
> 
> image

Hello ! Thanks for your update :muscle: 

Here is a screenshot of the post from Roman :

image

Could you provide me your Discord ID so we can add you to the channel ?

**Comment by sponomarev** [comment]
Thank you @hedi-edelbloute. 🙏  I've already got in touch with Roman. We will figure it out the process and the requirements with him.

**Comment by hedi-edelbloute** [comment]
Good morning @sponomarev , ping me when you think I should pass again on your PR

Thanks for your work have a nice weekend

**Comment by hedi-edelbloute** [comment]
Hey @sponomarev, is your work done here ? Do I need to test/review it ?

**Comment by hedi-edelbloute** [comment]
Hey @sponomarev , do i need to review and test your PR ? Have a nice day

**Comment by sponomarev** [comment]
Hey @hedi-edelbloute! Seems that we're ready to go. Looking forward to seeing your feedback.

**Comment by hedi-edelbloute** [comment]
Hello, @sponomarev , review in progress :handshake: ! Thanks for your work

**Review by hedi-edelbloute — APPROVED** [review]

**Comment by sponomarev** [comment]
- ✅ typecheck fix
- ✅ rebase over the latest master

File: libs/ledger-live-common/src/families/solana/types.ts [hunk 1]
```diff
@@ -212,10 +213,12 @@ export type SolanaStakeWithMeta = {
 
 export type SolanaResources = {
   stakes: SolanaStake[];
+  unstakeReserve: BigNumber;
 };
 
 export type SolanaResourcesRaw = {
   stakes: string;
+  unstakeReserve: BigNumber;
```
Comment (haammar-ledger) [inline libs/ledger-live-common/src/families/solana/types.ts:221]: `*Raw` types must only contain native types (i.e compatible with `JSON.stringify`)

Please use `string` here and update the functions in `serialization.ts`

**Review by haammar-ledger — CHANGES_REQUESTED** [review]

File: libs/ledger-live-common/src/families/solana/types.ts (hunk 1 above)
Comment (sponomarev) [reply libs/ledger-live-common/src/families/solana/types.ts]: Sure! Fixed in [9ef14bc](https://github.com/LedgerHQ/ledger-live/pull/5228/commits/9ef14bcbe9bb1741142803b1df8bfb96fd4d7b01). Just curious, what is the purpose for this serialization?

Comment (hedi-edelbloute) [reply libs/ledger-live-common/src/families/solana/types.ts]: Account is saved in the cache, this method transform the actual "living" object in storable data, we want only native types there

**Comment by hedi-edelbloute** [comment]
Hey @sponomarev, hope you are well

I cloned the PR on our repo here : https://github.com/LedgerHQ/ledger-live/pull/5797 to run our transaction bot and QA

Will come back to you when it is finished

**Comment by sponomarev** [comment]
@hedi-edelbloute Great! Let me know if you need anything from our side.

**Comment by sponomarev** [comment]
@hedi-edelbloute @haammar-ledger Hey there! Happy holidays! Is anything else needed from our side to get that merged?

**Comment by hedi-edelbloute** [comment]
Hey @sponomarev, i am off this week and coming back on january 2nd

I ran the bot multiple times, synchronisation timeouts but i dont know yet if it's a configuration issue or related to the update itself

Happy new year's eve

**Comment by sponomarev** [comment]
- rebased over the latest develop
- the bot is not very stable even locally. I face with Solana RPC timeouts not only in tests, but in regular CLI hassle also. Nevertheless, I was able to pass it through.

```
$ pnpm run:cli bot -c solana

> ledger-live@0.0.0 run:cli /Users/toad/Projects/Hoodies/Solana/ledger-live
> ./apps/cli/bin/index.js "bot" "-c" "solana"

necessary accounts resynced in 0.16ms
▬ Solana 1.2.0 on nanoS 2.0.0
→ FROM Solana 1 cross: 0.0254521 SOL (27ops) (HqHkGp32W1ZsDRcBmYYXVzvTFrSNd16sWGbR6ZVdACCZ on 44'/501'/0') solanaSub#0 js:2:solana:HqHkGp32W1ZsDRcBmYYXVzvTFrSNd16sWGbR6ZVdACCZ:solanaSub (! sum of ops 0.02306929 SOL)
max spendable ~0.0221684
★ using mutation 'Transfer ~50%'
→ TO Solana 8: 0.00415471 SOL (1ops) (HTMmzoJ6uYJvvwANhgmUvsfbLzi8z4YneYeJHEM8xPkh on 44'/501'/7') solanaSub#7 js:2:solana:HTMmzoJ6uYJvvwANhgmUvsfbLzi8z4YneYeJHEM8xPkh:solanaSub
✔️ transaction
  SEND: 0.011619852 SOL
  TO: HTMmzoJ6uYJvvwANhgmUvsfbLzi8z4YneYeJHEM8xPkh
  MEMO: a memo
STATUS (790ms)
  amount: 0.011619852 SOL
  estimated fees: 0.000005 SOL
  total spent: 0.011624852 SOL
errors:
errors:
✔️ has been signed! (2220ms)
✔️ broadcasted! (8.7s) optimistic operation:
  -0.011624852 SOL   OUT        5hHoDT3xwnpQEx1k7MyBHoPF6pfK8hKZ62fExrTfnDr2Vs7j95icFbASi6bFTWzMXh9Dd78GNs5wKWdbTeBxiUyn 2023-12-28T17:19
✔️ operation confirmed (38.8s):
  -0.011624852 SOL   OUT        5hHoDT3xwnpQEx1k7MyBHoPF6pfK8hKZ62fExrTfnDr2Vs7j95icFbASi6bFTWzMXh9Dd78GNs5wKWdbTeBxiUyn 2023-12-28T17:19
✔️ Solana 1 cross: 0.0138273 SOL (28ops) (HqHkGp32W1ZsDRcBmYYXVzvTFrSNd16sWGbR6ZVdACCZ on 44'/501'/0') solanaSub#0 js:2:solana:HqHkGp32W1ZsDRcBmYYXVzvTFrSNd16sWGbR6ZVdACCZ:solanaSub (! sum of ops 0.011444438 SOL)(in 38.8s)
✔️ destination operation
  +0.011619852 SOL   IN         5hHoDT3xwnpQEx1k7MyBHoPF6pfK8hKZ62fExrTfnDr2Vs7j95icFbASi6bFTWzMXh9Dd78GNs5wKWdbTeBxiUyn 2023-12-28T17:19
(in 20.1s)

necessary accounts resynced in 0.18ms
▬ Solana 1.2.0 on nanoS 2.0.0
→ FROM Solana 2: 0.00089088 SOL (26ops) (Enoich98Ck4cSrAs8xsH9TecNf4fnYwdwR3pHKM9NkFv on 44'/501'/1') solanaSub#1 js:2:solana:Enoich98Ck4cSrAs8xsH9TecNf4fnYwdwR3pHKM9NkFv:solanaSub
max spendable ~0
🤷‍♂️ couldn't find a mutation to do! (Transfer ~50%: balance is 0, Transfer Max: balance is 0, Delegate: not enough balance, Deactivate Activating Delegation: not enough balance, Deactivate Active Delegation: not enough balance, Reactivate Deactivating Delegation: not enough balance, Activate Inactive Delegation: not enough balance, Withdraw Delegation: not enough balance)

necessary accounts resynced in 0.15ms
▬ Solana 1.2.0 on nanoS 2.0.0
→ FROM Solana 3: 0.0100515 SOL (29ops) (BCfXbxB83GRzYGkvQ7Rjh6rmHLTrKtjpbtn9PPUGGR4j on 44'/501'/2') solanaSub#2 js:2:solana:BCfXbxB83GRzYGkvQ7Rjh6rmHLTrKtjpbtn9PPUGGR4j:solanaSub (! sum of ops 0.007668695 SOL)
max spendable ~0.00676281
★ using mutation 'Transfer Max'
→ TO Solana 5: 0.00936729 SOL (18ops) (d3P5kjxsy1MMh2TSgiiT6nHsqa6uwDHtv8ocYpxyYtb on 44'/501'/4') solanaSub#4 js:2:solana:d3P5kjxsy1MMh2TSgiiT6nHsqa6uwDHtv8ocYpxyYtb:solanaSub
✔️ transaction
  SEND: 0.006762815 SOL (ALL)
  TO: d3P5kjxsy1MMh2TSgiiT6nHsqa6uwDHtv8ocYpxyYtb
  MEMO: a memo
STATUS (813ms)
  amount: 0.006762815 SOL
  estimated fees: 0.000005 SOL
  total spent: 0.006767815 SOL
errors:
errors:
✔️ has been signed! (1924ms)
✔️ broadcasted! (25.7s) optimistic operation:
  -0.006767815 SOL   OUT        4Bf9Mt31zrwxRdb7DA3HRuocyPQLyrpTPAJWk5FwPZ8fHZp15teUnCVpZ572JgepiiHYHDWmVt7kHEPFjbZ6o94x 2023-12-28T17:20
✔️ operation confirmed (37.9s):
  -0.006767815 SOL   OUT        4Bf9Mt31zrwxRdb7DA3HRuocyPQLyrpTPAJWk5FwPZ8fHZp15teUnCVpZ572JgepiiHYHDWmVt7kHEPFjbZ6o94x 2023-12-28T17:21
✔️ Solana 3: 0.00328376 SOL (30ops) (BCfXbxB83GRzYGkvQ7Rjh6rmHLTrKtjpbtn9PPUGGR4j on 44'/501'/2') solanaSub#2 js:2:solana:BCfXbxB83GRzYGkvQ7Rjh6rmHLTrKtjpbtn9PPUGGR4j:solanaSub (! sum of ops 0.00090088 SOL)(in 37.9s)

necessary accounts resynced in 10.6s
▬ Solana 1.2.0 on nanoS 2.0.0
→ FROM Solana 4: 0.00089088 SOL (16ops) (2qEMgNSnmSZ6F9yvQFP81dccsHsrCXf5xdNVi97gG7EE on 44'/501'/3') solanaSub#3 js:2:solana:2qEMgNSnmSZ6F9yvQFP81dccsHsrCXf5xdNVi97gG7EE:solanaSub
max spendable ~0
🤷‍♂️ couldn't find a mutation to do! (Transfer ~50%: balance is 0, Transfer Max: maximum mutation run reached (1), Delegate: not enough balance, Deactivate Activating Delegation: not enough balance, Deactivate Active Delegation: not enough balance, Reactivate Deactivating Delegation: not enough balance, Activate Inactive Delegation: not enough balance, Withdraw Delegation: not enough balance)

necessary accounts resynced in 0.19ms
▬ Solana 1.2.0 on nanoS 2.0.0
→ FROM Solana 5: 0.0161301 SOL (19ops) (d3P5kjxsy1MMh2TSgiiT6nHsqa6uwDHtv8ocYpxyYtb on 44'/501'/4') solanaSub#4 js:2:solana:d3P5kjxsy1MMh2TSgiiT6nHsqa6uwDHtv8ocYpxyYtb:solanaSub
max spendable ~0.0152342
★ using mutation 'Transfer ~50%'
→ TO Solana 9: 0 SOL (0ops) (12tDGeEACr6tnBQu691XSMPfcdXnKBM1BPpWiGCgSgmn on 44'/501'/8') solanaSub#8 js:2:solana:12tDGeEACr6tnBQu691XSMPfcdXnKBM1BPpWiGCgSgmn:solanaSub
✔️ transaction
  SEND: 0.007407628 SOL
  TO: 12tDGeEACr6tnBQu691XSMPfcdXnKBM1BPpWiGCgSgmn
STATUS (821ms)
  amount: 0.007407628 SOL
  estimated fees: 0.000005 SOL
  total spent: 0.007412628 SOL
errors:
errors: recipient SolanaAccountNotFunded
✔️ has been signed! (1957ms)
✔️ broadcasted! (2919ms) optimistic operation:
  -0.007412628 SOL   OUT        J85gzTZWAeDkAjdRgFsvy1YY7Zon8GCE5m7EeadXHebMoWcrMr4byGymHEmQkRYNnBczKE7pM39crKkdsgFdp5c 2023-12-28T17:22
✔️ operation confirmed (30.5s):
  -0.007412628 SOL   OUT        J85gzTZWAeDkAjdRgFsvy1YY7Zon8GCE5m7EeadXHebMoWcrMr4byGymHEmQkRYNnBczKE7pM39crKkdsgFdp5c 2023-12-28T17:22
✔️ Solana 5: 0.00871748 SOL (20ops) (d3P5kjxsy1MMh2TSgiiT6nHsqa6uwDHtv8ocYpxyYtb on 44'/501'/4') solanaSub#4 js:2:solana:d3P5kjxsy1MMh2TSgiiT6nHsqa6uwDHtv8ocYpxyYtb:solanaSub(in 30.5s)
✔️ destination operation
  +0.007407628 SOL   IN         J85gzTZWAeDkAjdRgFsvy1YY7Zon8GCE5m7EeadXHebMoWcrMr4byGymHEmQkRYNnBczKE7pM39crKkdsgFdp5c 2023-12-28T17:22
(in 20.1s)

necessary accounts resynced in 0.71ms
▬ Solana 1.2.0 on nanoS 2.0.0
→ FROM Solana 6: 0.0201511 SOL (13ops) (3pA7WPGFroPFwt5zNAztaY93VLVEQTAM1VjmuoDiZwTY on 44'/501'/5') solanaSub#5 js:2:solana:3pA7WPGFroPFwt5zNAztaY93VLVEQTAM1VjmuoDiZwTY:solanaSub (! sum of ops 0.015385392 SOL)
max spendable ~0.0144695
★ using mutation 'Delegate'
✔️ transaction
  CREATE STAKE ACCOUNT: 73iErZLetySyjR7EnjATcnjLDbGhuCvUibB2vdGvgNoJ
  FROM: 3pA7WPGFroPFwt5zNAztaY93VLVEQTAM1VjmuoDiZwTY
  AMOUNT: 0.00238288 SOL
  SEED: stake:0.4965632098186734
  VALIDATOR: 3CnKZPQn92W8WXG7KTVaFQRk8LJJ3KZbrVVF4ngUxqkg
STATUS (4.3s)
  amount: 0.0001 SOL
  estimated fees: 0.00228788 SOL
  total spent: 0.00238788 SOL
errors:
errors:
✔️ has been signed! (2263ms)
✔️ broadcasted! (4.5s) optimistic operation:
  -0.00238788 SOL    DELEGATE   Y7ni2KX9GnBAZ4NnvR6NpXCqmNXsjCF3ARihnxVpZyDpJgGJ59eAdLWDbCYDhhRp9A6EJctH5d8M1dSooUv9zYJ 2023-12-28T17:23
✔️ operation confirmed (40.5s):
  -0.00238788 SOL    DELEGATE   Y7ni2KX9GnBAZ4NnvR6NpXCqmNXsjCF3ARihnxVpZyDpJgGJ59eAdLWDbCYDhhRp9A6EJctH5d8M1dSooUv9zYJ 2023-12-28T17:23
✔️ Solana 6: 0.0201461 SOL (14ops) (3pA7WPGFroPFwt5zNAztaY93VLVEQTAM1VjmuoDiZwTY on 44'/501'/5') solanaSub#5 js:2:solana:3pA7WPGFroPFwt5zNAztaY93VLVEQTAM1VjmuoDiZwTY:solanaSub (! sum of ops 0.012997512 SOL)(in 40.5s)

necessary accounts resynced in 0.14ms
▬ Solana 1.2.0 on nanoS 2.0.0
→ FROM Solana 7: 0.00869137 SOL (5ops) (EZHser4BYx8zpNCq8ZpvB3oCzxWicrooQQXiYWxRwrpB on 44'/501'/6') solanaSub#6 js:2:solana:EZHser4BYx8zpNCq8ZpvB3oCzxWicrooQQXiYWxRwrpB:solanaSub
max spendable ~0.00779549
🤷‍♂️ couldn't find a mutation to do! (Transfer ~50%: maximum mutation run reached (2), Transfer Max: maximum mutation run reached (1), Delegate: maximum mutation run reached (1), Deactivate Activating Delegation: no activating stakes found, Deactivate Active Delegation: no active stakes found, Reactivate Deactivating Delegation: no deactivating stakes found, Activate Inactive Delegation: no inactive stakes found, Withdraw Delegation: no withdrawable stakes found)

necessary accounts resynced in 0.11ms
▬ Solana 1.2.0 on nanoS 2.0.0
→ FROM Solana 8: 0.0157745 SOL (2ops) (HTMmzoJ6uYJvvwANhgmUvsfbLzi8z4YneYeJHEM8xPkh on 44'/501'/7') solanaSub#7 js:2:solana:HTMmzoJ6uYJvvwANhgmUvsfbLzi8z4YneYeJHEM8xPkh:solanaSub
max spendable ~0.0148786
🤷‍♂️ couldn't find a mutation to do! (Transfer ~50%: maximum mutation run reached (2), Transfer Max: maximum mutation run reached (1), Delegate: maximum mutation run reached (1), Deactivate Activating Delegation: no activating stakes found, Deactivate Active Delegation: no active stakes found, Reactivate Deactivating Delegation: no deactivating stakes found, Activate Inactive Delegation: no inactive stakes found, Withdraw Delegation: no withdrawable stakes found)

necessary accounts resynced in 0.83ms
▬ Solana 1.2.0 on nanoS 2.0.0
→ FROM Solana 9: 0.00740762 SOL (1ops) (12tDGeEACr6tnBQu691XSMPfcdXnKBM1BPpWiGCgSgmn on 44'/501'/8') solanaSub#8 js:2:solana:12tDGeEACr6tnBQu691XSMPfcdXnKBM1BPpWiGCgSgmn:solanaSub
max spendable ~0.00651174
🤷‍♂️ couldn't find a mutation to do! (Transfer ~50%: maximum mutation run reached (2), Transfer Max: maximum mutation run reached (1), Delegate: maximum mutation run reached (1), Deactivate Activating Delegation: no activating stakes found, Deactivate Active Delegation: no active stakes found, Reactivate Deactivating Delegation: no deactivating stakes found, Activate Inactive Delegation: no inactive stakes found, Withdraw Delegation: no withdrawable stakes found)
```

**Pull request merged**